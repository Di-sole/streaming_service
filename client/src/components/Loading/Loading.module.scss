.container {
    position: relative;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    color: rgba(255, 255, 255, 0.48);
    font-size: 16px;
    margin: 60px 0px 20px;
}
.spinner {
    width: 50px;
    height: 50px;
    transform: translate(-50%, -50%);
    margin-left: 40px;

    .blob {
        position: absolute;
        top: 50%;
        left: 50%;
        transform: translate(-50%, -50%);
        border: 2px solid rgba(255, 255, 255, 0.48);

        width: 15px;
        height: 15px;
        border-radius: 50%;

        &.top {
            top: 0;
            animation: blob-top 1s infinite ease-in;
        }
        &.bottom {
            top: 100%;
            animation: blob-bottom 1s infinite ease-in;
        }
        &.left {
            left: 0;
            animation: blob-left 1s infinite ease-in;
        }
    }

    .move {
        background: rgba(255, 255, 255, 0.78);
        top: 0;

        animation: blob-spinner-mover 1s infinite ease-in;
    }
}

@keyframes blob-bottom {
    25%,
    50%,
    75% {
        top: 50%;
        left: 100%;
    }
    100% {
        top: 0;
        left: 50%;
    }
}
@keyframes blob-left {
    25% {
        top: 50%;
        left: 0;
    }
    50%,
    100% {
        top: 100%;
        left: 50%;
    }
}
@keyframes blob-top {
    50% {
        top: 0;
        left: 50%;
    }
    75%,
    100% {
        top: 50%;
        left: 0;
    }
}

@keyframes blob-spinner-mover {
    0%,
    100% {
        top: 0;
        left: 50%;
    }
    25% {
        top: 50%;
        left: 100%;
    }
    50% {
        top: 100%;
        left: 50%;
    }
    75% {
        top: 50%;
        left: 0;
    }
}
