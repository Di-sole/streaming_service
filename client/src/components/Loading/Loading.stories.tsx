import { Meta, StoryObj } from '@storybook/react';
import './Loading.module.scss';
import Load from './Loading';

const meta: Meta<typeof Load> = {
    title: 'Information/Loading',
    component: Load,
};

export default meta;

type Story = StoryObj<typeof Load>;

export const Loading: Story = {};
