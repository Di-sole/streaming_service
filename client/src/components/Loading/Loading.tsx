import { useTranslation } from 'next-i18next';
import styles from './Loading.module.scss';

const Loading = () => {
    const { t } = useTranslation('moviesPage');
    return (
        <div className={styles.container} data-testid={'loading'}>
            <div className={styles.spinner}>
                <div className={[styles.blob, styles.top].join(' ')}></div>
                <div className={[styles.blob, styles.bottom].join(' ')}></div>
                <div className={[styles.blob, styles.left].join(' ')}></div>

                <div className={[styles.blob, styles.move].join(' ')}></div>
            </div>
            {t('loading')}
        </div>
    );
};

export default Loading;
