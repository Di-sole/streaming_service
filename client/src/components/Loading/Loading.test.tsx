import { render, screen } from '@testing-library/react';
import Loading from './Loading';

jest.mock('next-i18next', () => ({
    useTranslation: () => {
        return {
            t: (str: string) => str,
            i18n: {
                changeLanguage: () => new Promise(() => {}),
            },
        };
    },
}));

jest.mock('next/router', () => ({
    useRouter() {
        return {
            pathname: '',
        };
    },
}));

describe('LOADING SNAPSHOTS TESTS', () => {
    test('Loading should not be changed', () => {
        render(<Loading />);

        const id = screen.getByTestId('loading');
        expect(id).toMatchSnapshot();
    });
});
