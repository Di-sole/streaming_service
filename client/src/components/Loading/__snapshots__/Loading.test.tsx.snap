// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LOADING SNAPSHOTS TESTS Loading should not be changed 1`] = `
<div
  class="container"
  data-testid="loading"
>
  <div
    class="spinner"
  >
    <div
      class="blob top"
    />
    <div
      class="blob bottom"
    />
    <div
      class="blob left"
    />
    <div
      class="blob move"
    />
  </div>
  loading
</div>
`;
